fineract-provider/src/main/java/org/apache/fineract/portfolio/search/service/SearchReadPlatformServiceImpl.java

```diff
@@ -90,7 +90,7 @@ public String searchSchema(final SearchConditions searchConditions) {
                 + " order by l.id desc";
 
         final String savingMatchSql = " (select 'SAVING' as entityType, s.id as entityId, sp.name as entityName, s.external_id as entityExternalId, s.account_no as entityAccountNo "
-                + " , coalesce(c.id,g.id) as parentId, coalesce(c.display_name,g.display_name) as parentName, null as entityMobileNo, s.status_enum as entityStatusEnum, s.deposit_type_enum as subEntityType, CASE WHEN g.id is null THEN 'client' ELSE 'group' END as parentType "
+                + " , coalesce(c.id,g.id) as parentId, coalesce(c.display_name, g.display_name) as parentName, null as entityMobileNo, s.status_enum as entityStatusEnum, concat(s.deposit_type_enum, '') as subEntityType, CASE WHEN g.id is null THEN 'client' ELSE 'group' END as parentType "
```

Comment:
Every other places the `subEntityType` is null, i dont think thats the place to fix this. Please rather have a condition in the mapper to avoid NPE issues.

Reply:
Nothing changed...

Reply:
how does solve this problem by changing this to not be null only one place? what happens if you search in SAVINGS accounts and LOAN accounts?